src/include/module/module/system_service.h

```diff
@@ -0,0 +1,113 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright(c) 2024 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Warecki <adrian.warecki@intel.com>
+ */
+
+#ifndef MODULE_MODULE_SYSTEM_SERVICE_H
+#define MODULE_MODULE_SYSTEM_SERVICE_H
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include "logger.h"
+#include "../iadk/adsp_error_code.h"
+
+#ifdef __XTENSA__
+	#define RESTRICT __restrict
+#else
+	#define RESTRICT
+#endif
```

Comment:
It may be best to comment why this cant come from Zephyr as I suspect Zephyr will define this too.

Reply:
ping - this needs a comment at least so that no one tries to optimize this later on.

Reply:
@lgirdwood 
Indeed zephyr has a similar definition `ZRESTRICT`, but we want to be able to build lmdk modules without using zephyr headers so I didn't use it.